org.civicrm.reqangular/src/common/services/api/contract.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-env amd */
+
+define([
+  'common/modules/apis',
+  'common/services/api'
+], function (apis, _) {
+  'use strict';
+
+  apis.factory('api.contract', ['$log', 'api', function ($log, api) {
+    $log.debug('api.contract');
+
+    return api.extend({
+
+      /**
+       * Returns all contracts.
+       *
+       * It chains an additional call to the `getfulldetails` endpoint to also return
+       * full details on contracts
+       *
+       * @param  {Object} params  matches the api endpoint params (ex. contact_id)
+       * @return {Promise}
+       */
+      all: function (params) {
+        $log.debug('ContractAPI.all');
+
+        params = params || {};
+
+        params['api.HRJobContract.getfulldetails'] = { 'jobcontract_id': '$value.id' };
+        params['deleted'] = 0;
+
+        return this.sendGET('HRJobContract', 'get', params, false)
+          .then(function (data) {
+            return data.values;
```

Comment:
When the chained API returns the values, instead of keeping those in the same key('api.HRJobContract.getfulldetails') change it to a more simple name like `details`. It is easier to understand and manage.
For reference: https://github.com/civicrm/civihr/blob/feb622033ced46b16dba5409e3638d13dccdafe8/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/entitlement-api.js#L82-L82

Similar stuff was done here.